### DISCUSSION TITLE
[Run issue]: Issues with Running OpenTOPAS on macOS: Qt "cocoa" Plugin Not Found

### DISCUSSION THREAD

**Discussion by HussainAther** [OpenTOPAS/OpenTOPAS#78] [Run-Problem]
### Contact Details

shussainather@gmail.com

### What was the issue?

Description:

I have been attempting to run OpenTOPAS on macOS (Sonoma), but I keep encountering issues related to the Qt "cocoa" plugin. Despite following multiple troubleshooting steps, the problem persists. The error I consistently receive is:

qt.qpa.plugin: Could not find the Qt platform plugin "cocoa" in "/opt/homebrew/opt/qt/plugins/platforms"  
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.  
zsh: abort topas Isotope.txt  

Steps I’ve Taken:  
1. Installed OpenTOPAS 4.0 and confirmed it is compatible with macOS.  
2. Installed Qt 5 (version 5.15.11) using Homebrew at `/opt/homebrew/opt/qt@5`.  
3. Verified that the file `libqcocoa.dylib` is missing from `/opt/homebrew/opt/qt@5/plugins/platforms`.  
4. Set environment variables explicitly pointing to the Qt installation:  

```bash  
export QT_QPA_PLATFORM_PLUGIN_PATH=/opt/homebrew/opt/qt@5/plugins/platforms  
export QT_PLUGIN_PATH=/opt/homebrew/opt/qt@5/plugins  
```  
5. Tried installing `libqcocoa.dylib` manually, but still receiving the same error.  
6. Tested with GUI disabled in OpenTOPAS (b:Ts/UseQt = "False"), which works, but I need the GUI for my simulations.  
7. Removed any potential Qt version conflicts (such as `qt@6`) using Homebrew.  

Setup:  
- macOS Version: Sonoma  
- OpenTOPAS Version: 4.0  
- Qt Version: 5.15.11 (installed via Homebrew)  
- Error: Qt "cocoa" plugin not found, preventing GUI from initializing.  

Error Log:  
Welcome to TOPAS, Tool for Particle Simulation (Version 3.9)  
Loading parameters starting from: Isotope.txt  
qt.qpa.plugin: Could not find the Qt platform plugin "cocoa" in "/opt/homebrew/opt/qt/plugins/platforms"  
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.  
zsh: abort topas Isotope.txt  

Request:  
Does anyone have experience with resolving this issue on macOS or can suggest alternative methods for getting the Qt "cocoa" plugin to work with OpenTOPAS? I would appreciate any guidance or suggestions. I have been struggling with this issue for over a month now.

Please let me know if you need further details or if there is another version of Qt that would be more suitable for OpenTOPAS on macOS.  

Thank you for your help!

### TOPAS category

Geometry

### Version

4.0

### What Operative System (OS) are you running on?

MacOS

### Relevant log output

```shell
Welcome to TOPAS, Tool for Particle Simulation (Version 3.9)
Loading parameters starting from: Isotope.txt
qt.qpa.plugin: Could not find the Qt platform plugin "cocoa" in "/opt/homebrew/opt/qt/plugins/platforms"
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
zsh: abort      topas Isotope.txt
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by icmeyer** [comment]
Were ever able to resolve this issue? Looks like you already tried a lot!

Did you try this [SO post](https://stackoverflow.com/questions/54297627/qt-could-not-find-the-platform-plugin-cocoa)?

**Comment by JoseRamosMendez** [comment]
Hi there,
The TOPAS welcome message states you're using version 3.9, would you try with version 4.0 https://github.com/OpenTOPAS/OpenTOPAS?
Best.
